torch/_refs/nn/functional/__init__.py

```diff
@@ -59,6 +60,43 @@
 
 Tensor = torch.Tensor
 
+
+@register_decomposition(torch.ops.aten.alpha_dropout)
+def alpha_dropout(
+    self: TensorLikeType, p: float = 0.5, training: bool = True, inplace: bool = False
+) -> TensorLikeType:
+
+    if inplace:
+        raise NotImplementedError
+
+    if not training:
+        return self
+
+    assert p <= 1
+    assert p >= 0
+
+    if p == 1:
+        return refs.zeros_like(self)
+
+    if p == 0:
+        return self
+
+    noise = refs.lt(
```

Comment:
Prefer `torch` operations when possible, and only use refs explicitly if there's some needed functionality the ref offers that the corresponding torch operation doesn't

Reply:
Instead of `noise`, this might more accurately be described as the `dropout_mask`?

Reply:
Hi, @mruberry! Thanks for your input. I will keep this in my mind for the upcoming PRs.

In the latest commit, have used torch operators and changed the name to `dropout_mask`.